### PR TITLE
Better HoltWinters parameter validation

### LINKED ISSUE #34046: [Test] MovAvgIT#testBadModelParams needs clarification
Currently the "testBadModelParams" method does seem to pass regardless of whether bad
model parameters are used or not. My guess is that its supposed to be testing that an exception is 
thrown if you provide invalid model parameters. From a conversation with @colings86 we think that
the padding parameter passed in ensure the alpha and beta values are invalid, but that seems to be
trappy because if the randomness selects simple or linear models the padding isn’t used and the model 
will be valid. I will @AwaitsFix the test in the course of https://github.com/elastic/elasticsearch/pull/31876
where I stumbled into it, but create this issue for further investigation.

### PR DESCRIPTION
**Pull request by polyfractal**
We validate HW parameters (namely, `window > 2 * period`) when parsing the XContent... but that means transport clients can configure bad params.

This adds validation to when configuring the model or window on the builder, and in `doValidate()`.  Might be overkill to also add it to `doValidate()` but figured it couldn't hurt and would help in heterogeneous clusters.

It also makes some test changes:

- removes testBadModelParams(), which was a junk test (didn't do
anything), and bad param checking is done elsewhere in units tests
- Fixes one of the windows in testHoltWintersNotEnoughData()
- Ensures the period in testHoltWintersNotEnoughData() is >> window
- Removes `setTypes()` since that's deprecated

Closes #34046

**NOTE:** this targets 7.x because MovAvg is deprecated and will be removed from master shortly

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgPipelineAggregationBuilder.java [hunk 1]
```diff
@@ -147,6 +147,7 @@ public MovAvgPipelineAggregationBuilder window(int window) {
         if (window <= 0) {
             throw new IllegalArgumentException("[window] must be a positive integer: [" + name + "]");
         }
+        HoltWintersModel.validateHWParams(model, window);
```
Comment (colings86) [inline server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgPipelineAggregationBuilder.java]: Unless I'm missing something, all the validation in this class will be applied no matter the model used. Should we not be only applying the validation if it's holt-winters? Maybe we should have a (non-static) validate method on the model class and this class just calls validate on the model?

**Review by colings86 — CHANGES_REQUESTED** [review]
I left a comment. Also wanted to check: is the new implementation of moving functions free from this issue?

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgPipelineAggregationBuilder.java (hunk 1 above)
Comment (polyfractal) [reply server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgPipelineAggregationBuilder.java]: The validate method makes sure the model is the correct type... but yeah it's sorta the wrong assignment of responsibilities (did it that way to minimize the number of `instanceof` checks scattered around).

++ to a validate method on the model itself, will move things around

**Comment by polyfractal** [comment]
> is the new implementation of moving functions free from this issue?

The new moving fn is indeed free from this issue.  Since the new version is all scripted, the check is done inside the HW function that's exposed from the context, so there's no similar issue with validating xcontent vs TC, etc.

It does mean we have to execute the script to find out the values provided by the user are bad... but that's just a general scripting issue :/

**Comment by polyfractal** [comment]
@colings86 think this is ready for a second look whenever you have a moment, no rush.

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgModel.java [hunk 2]
```diff
@@ -99,6 +99,15 @@ public boolean hasValue(int valuesAvailable) {
      */
     protected abstract double[] doPredict(Collection<Double> values, int numPredictions);
 
+    /**
+     * This method allows models to validate the window size if required
+     */
+    protected void validate(long window) {
```
Comment (colings86) [inline server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgModel.java]: Should we pass the agg name into this method so we can add it to the exceptions it throws?

**Review by colings86 — APPROVED** [review]
I left one minor comment but it LGTM

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgModel.java (hunk 2 above)
Comment (polyfractal) [reply server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovAvgModel.java]: :+1:

**Pull request merged**